bin/yaml/tools.yaml

```diff
@@ -107,3 +107,22 @@ tools:
     check_exe: bin/pahole
     targets:
       - trunk
+  iwyu:
```

Comment:
I'd really prefer us building this in the misc builder (once we extract the other things from there) else ... we will build an Ubuntu 22.04 (the infra node is 22.04) which probably won't work on the 20.04 nodes without some extra LD_LIBRARY path magic.